src/main/java/com/exasol/adapter/jdbc/BaseColumnMetadataReader.java

```diff
@@ -75,16 +75,19 @@ public List<ColumnMetadata> mapColumns(final String tableName) {
 
     /**
      * Read the columns metadata from a result set.
-     * 
+     *
      * @param catalogName catalog name
      * @param schemaName  schema name
      * @param tableName   table name
      * @return list with column metadata
      */
     protected List<ColumnMetadata> mapColumns(final String catalogName, final String schemaName,
             final String tableName) {
-        try (final ResultSet remoteColumns = this.connection.getMetaData().getColumns(catalogName, schemaName,
-                tableName, ANY_COLUMN)) {
+        final String tableNameEscaped = Wildcards.escape(tableName);
+        final String catalogNameEscaped = catalogName == null ? null : Wildcards.escape(catalogName);
+        final String schemaNameEscaped = schemaName == null ? null : Wildcards.escape(schemaName);
+        try (final ResultSet remoteColumns = this.connection.getMetaData().getColumns(catalogNameEscaped,
+                schemaNameEscaped, tableNameEscaped, ANY_COLUMN)) {
```

Comment:
Extract call to `this.connection.getMetaData().getColumns(...)` incl. escaping into a method that returns a ResultSet.